Gems/Atom/Tools/MaterialEditor/Code/Source/Document/MaterialDocument.cpp

```diff
@@ -608,89 +362,64 @@ namespace MaterialEditor
             return true;
         });
 
-        return result;
-    }
-
-    bool MaterialDocument::OpenInternal(AZStd::string_view loadPath)
-    {
-        using namespace AZ;
-        using namespace RPI;
-
-        Clear();
-
-        m_absolutePath = loadPath;
-        if (!AzFramework::StringFunc::Path::Normalize(m_absolutePath))
+        if (!addPropertiesResult || !AZ::RPI::JsonUtils::SaveObjectToFile(m_savePathNormalized, sourceData))
```

Comment:
nit: this `if` line does 2 error checkings and also saves at the same time, and the order matters. Consider splitting them in 2 to make it more explicit and put specific errors on each one.

````
if (!addPropertiesResult)
{
    AZ_Error("MaterialDocument", false, "Properties were not added correctly");
    return false;
}

// Save object
bool savedSuccessfully = AZ::RPI::JsonUtils::SaveObjectToFile(m_savePathNormalized, sourceData));
AZ_Error("MaterialDocument", savedSuccessfully, "Document could not be saved: '%s'.", m_savePathNormalized.c_str());

return savedSuccessfully;
````